[CI] GenerativeRandomMappingIT test fails

### CI Link

https://gradle-enterprise.elastic.co/s/gu3ao3k7k63ga/tests/task/:x-pack:plugin:esql:qa:server:single-node:javaRestTest/details/org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT/test?top-execution=1

### Repro line

`./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT.test" -Dtests.seed=DE2E898295190233 -Dtests.locale=tn -Dtests.timezone=America/Port-au-Prince -Druntime.java=26`

### Does it reproduce?

No

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

Only fails on ARM

```
GenerativeRandomMappingIT > test FAILED
    java.lang.AssertionError: query: from gen_*,gen_rm_1
    exception: method [POST], host [http://[::1]:39419], URI [/_query?format=yaml&pretty=true&error_trace=true], status line [HTTP/1.1 500 Internal Server Error]
    Warnings: [No limit defined, adding default limit of [1000]]
    ---
    error:
      root_cause:
      - type: "illegal_state_exception"
        reason: "Cannot load blocks without doc values"
        stack_trace: "org.elasticsearch.ElasticsearchException$1: Cannot load blocks without\
          \ doc values\n\tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:810)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:724)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.build(RestResponse.java:202)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:161)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:122)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:54)\n\
          \tat org.elasticsearch.xpack.esql.action.EsqlResponseListener.lambda$wrapWithLogging$1(EsqlResponseListener.java:191)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:282)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.rest.action.RestCancellableNodeClient$1.onFailure(RestCancellableNodeClient.java:97)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:89)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:442)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:235)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:89)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:32)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:360)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:89)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:442)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:89)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:32)\n\
          \tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.lambda$innerExecute$9(TransportEsqlQueryAction.java:367)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:282)\n\
          \tat org.elasticsearch.xpack.esql.execution.PlanExecutor.onQueryFailure(PlanExecutor.java:164)\n\
          \tat org.elasticsearch.xpack.esql.execution.PlanExecutor.lambda$esql$1(PlanExecutor.java:134)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:282)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener$FailureResult.complete(SubscribableListener.java:448)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:351)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:376)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.onFailure(SubscribableListener.java:288)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:89)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:32)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener$FailureResult.complete(SubscribableListener.java:448)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:351)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:376)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.onFailure(SubscribableListener.java:288)\n\
          \tat org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$executePlan$11(ComputeService.java:628)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:202)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:207)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:89)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:32)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:360)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:89)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:442)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:78)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:89)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:32)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:263)\n\
          \tat org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$new$0(ComputeListener.java:37)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingFailureActionListener.onResponse(ActionListenerImplementations.java:233)\n\
          \tat org.elasticsearch.compute.EsqlRefCountingListener.lambda$new$0(EsqlRefCountingListener.java:31)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:125)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:77)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:122)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.Releasables$4.close(Releasables.java:181)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.Releasables.close(Releasables.java:39)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:72)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:51)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:234)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:432)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onResponse(ActionListenerImplementations.java:350)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:198)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onResponse(ActionListenerImplementations.java:350)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:198)\n\
          \tat org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$acquireCompute$2(ComputeListener.java:63)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:274)\n\
          \tat org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$executePlan$16(ComputeService.java:773)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingFailureActionListener.onResponse(ActionListenerImplementations.java:233)\n\
          \tat org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$new$0(ComputeListener.java:37)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingFailureActionListener.onResponse(ActionListenerImplementations.java:233)\n\
          \tat org.elasticsearch.compute.EsqlRefCountingListener.lambda$new$0(EsqlRefCountingListener.java:31)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:125)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:77)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:122)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.Releasables$4.close(Releasables.java:181)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.Releasables.close(Releasables.java:39)\n\
          \tat org.elasticsearch.base@9.5.0-SNAPSHOT/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:72)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:51)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:234)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:432)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onResponse(ActionListenerImplementations.java:350)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:198)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onResponse(ActionListenerImplementations.java:350)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:198)\n\
          \tat org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$acquireCompute$2(ComputeListener.java:63)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:274)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:111)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onResponse(ActionListenerImplementations.java:312)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:432)\n\
          \tat org.elasticsearch.compute.operator.DriverRunner$1.done(DriverRunner.java:65)\n\
          \tat org.elasticsearch.compute.operator.DriverRunner$1.onResponse(DriverRunner.java:43)\n\
          \tat org.elasticsearch.compute.operator.DriverRunner$1.onResponse(DriverRunner.java:40)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:274)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.TransportResponseHandler$1.handleResponse(TransportResponseHandler.java:55)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.TransportResponseHandler$Empty.handleResponse(TransportResponseHandler.java:45)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.TransportResponseHandler$Empty.handleResponse(TransportResponseHandler.java:37)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleResponse(TransportService.java:1746)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1524)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1622)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel$1.doRun(TransportService.java:1602)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:35)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1114)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\
          \tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)\n\
          \tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)\n\
          \tat java.base/java.lang.Thread.run(Thread.java:1516)\nCaused by: java.lang.IllegalStateException:\
          \ Cannot load blocks without doc values\n\tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.index.mapper.RangeFieldMapper$RangeFieldType.blockLoader(RangeFieldMapper.java:363)\n\
          \tat org.elasticsearch.xpack.esql.planner.EsPhysicalOperationProviders$DefaultShardContext.blockLoader(EsPhysicalOperationProviders.java:700)\n\
          \tat org.elasticsearch.xpack.esql.planner.EsPhysicalOperationProviders.blockLoaderAndConverter(EsPhysicalOperationProviders.java:254)\n\
          \tat org.elasticsearch.xpack.esql.planner.EsPhysicalOperationProviders.lambda$extractFields$7(EsPhysicalOperationProviders.java:513)\n\
          \tat org.elasticsearch.compute.lucene.read.ValuesSourceReaderOperator$FieldWork.newShard(ValuesSourceReaderOperator.java:421)\n\
          \tat org.elasticsearch.compute.lucene.read.ValuesSourceReaderOperator.positionFieldWork(ValuesSourceReaderOperator.java:335)\n\
          \tat org.elasticsearch.compute.lucene.read.ValuesFromSingleReader.loadFromSingleLeaf(ValuesFromSingleReader.java:87)\n\
          \tat org.elasticsearch.compute.lucene.read.ValuesFromSingleReader.load(ValuesFromSingleReader.java:59)\n\
          \tat org.elasticsearch.compute.lucene.read.ValuesReader.next(ValuesReader.java:46)\n\
          \tat org.elasticsearch.compute.lucene.read.ValuesReader.next(ValuesReader.java:26)\n\
          \tat org.elasticsearch.compute.operator.AbstractPageMappingToIteratorOperator$AppendBlocksIterator.next(AbstractPageMappingToIteratorOperator.java:342)\n\
          \tat org.elasticsearch.compute.operator.AbstractPageMappingToIteratorOperator$AppendBlocksIterator.next(AbstractPageMappingToIteratorOperator.java:318)\n\
          \tat org.elasticsearch.compute.operator.AbstractPageMappingToIteratorOperator$RuntimeTrackingIterator.next(AbstractPageMappingToIteratorOperator.java:194)\n\
          \tat org.elasticsearch.compute.operator.AbstractPageMappingToIteratorOperator$RuntimeTrackingIterator.next(AbstractPageMappingToIteratorOperator.java:179)\n\
          \tat org.elasticsearch.compute.operator.AbstractPageMappingToIteratorOperator.getOutput(AbstractPageMappingToIteratorOperator.java:153)\n\
          \tat org.elasticsearch.compute.operator.Driver.runSingleLoopIteration(Driver.java:323)\n\
          \tat org.elasticsearch.compute.operator.Driver.run(Driver.java:194)\n\tat org.elasticsearch.compute.operator.Driver$1.doRun(Driver.java:506)\n\
          \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\
          \tat org.elasticsearch.compute.operator.DriverScheduler$1.doRun(DriverScheduler.java:57)\n\
          \t... 7 more\n"
```

Another failure here https://gradle-enterprise.elastic.co/s/lpgba66tn4haq

```
REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT.test" -Dtests.seed=9BB8224B0BFF6589 -Dtests.locale=kw -Dtests.timezone=America/Lima -Druntime.java=26

GenerativeRandomMappingIT > test FAILED
    java.lang.AssertionError: query: from gen_*,gen_rm_3*,gen_rm_1
    exception: unexpected partial results: _clusters={details={(local)={_shards={total=4, failed=4, successful=0, skipped=0}, took=111, indices=gen_*,gen_rm_3*,gen_rm_1, failures=[{node=null, reason={reason=Cannot load blocks without doc values, type=illegal_state_exception}, index=gen_rm_2, shard=0}], status=partial}}}
    Expected: (null or is <false>)
         but: was <true>

    === Reproduction commands ===

    PUT /gen_rm_0
    {"mappings":{"properties": {"score": {"type": "histogram"},"tag": {"type": "float"},"category": {"type": "date"},"result": {"type": "double_range"},"level": {"type": "byte", "store": true, "index": false},"status": {"type": "date", "ignore_malformed": true, "locale": "fr"},"priority": {"type": "point"},"height": {"type": "half_float", "doc_values": false},"address": {"type": "long", "doc_values": false, "null_value": -18},"value": {"type": "long_range", "coerce": false, "index": false},"price": {"type": "integer", "ignore_malformed": true},"size": {"type": "integer"}}}}

    PUT /gen_rm_1
    {"mappings":{"properties": {"address": {"type": "long", "doc_values": false, "null_value": -18},"count": {"type": "integer", "doc_values": false, "ignore_malformed": true, "index": false},"result": {"type": "ip", "store": true},"height": {"type": "half_float"},"color": {"type": "long", "doc_values": false, "coerce": true},"weight": {"type": "text", "fields": {"raw": {"type": "keyword"}}},"price": {"type": "text", "store": true},"level": {"type": "byte", "store": true, "index": false},"active": {"type": "long", "store": true}}}}

    PUT /gen_rm_2
    {"mappings":{"properties": {"price": {"type": "token_count", "analyzer": "whitespace"},"code": {"type": "date_nanos"},"status": {"type": "date", "ignore_malformed": true, "locale": "fr", "doc_values": false},"value": {"type": "long"},"host": {"type": "float", "doc_values": false, "coerce": false},"score": {"type": "histogram"},"weight": {"type": "text", "fields": {"raw": {"type": "keyword"}}},"color": {"type": "long", "doc_values": false, "coerce": false},"height": {"type": "boolean"},"tag": {"type": "float", "doc_values": true, "coerce": true}}}}

    PUT /gen_rm_3
    {"mappings":{"properties": {"address": {"type": "double"},"weight": {"type": "text", "fields": {"raw": {"type": "keyword"}}},"category": {"type": "date"},"ver": {"type": "date_range", "doc_values": true},"region": {"type": "text", "store": true, "index_phrases": true},"rate": {"type": "date_range", "doc_values": false},"count": {"type": "rank_feature"},"code": {"type": "date_nanos"},"amount": {"type": "unsigned_long", "doc_values": true},"distance": {"type": "date"},"active": {"type": "long", "store": true, "doc_values": false},"size": {"type": "integer", "coerce": true},"path": {"type": "integer", "index": false},"name": {"type": "point"},"score": {"type": "search_as_you_type"}}}}

    POST /_bulk?refresh=true
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[89,7],"values":[-86.88,-66.85]},"tag":-63.22,"category":"2025-05-02T09:16:32.265Z","result":{"gte":-77.46,"lte":-16.339999999999996},"level":109,"priority":[-127.74,-945.87],"height":66.77,"address":938502,"value":{"gte":-564,"lte":1751},"size":430}
    {"index":{"_index":"gen_rm_0"}}
    {"tag":-0.92,"category":"2022-10-03T22:44:20.377Z","result":{"gte":-4.65,"lte":71.02},"level":51,"status":"2022-04-29T14:48:55.852Z","priority":[341.68,311.45],"height":53.91,"address":411630,"value":{"gte":-4396,"lte":1879},"price":831}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[84,96,46,99],"values":[-24.43,-21.64,0.63,37.21]},"category":"2023-09-23T07:38:03.455Z","result":[{"gte":-24.88,"lte":29.52},{"gte":11.45,"lte":87.07000000000001}],"level":75,"status":"2020-11-30T05:46:57.062Z","height":-80.39,"address":-223599,"value":{"gte":-9892,"lte":-8475},"price":854,"size":97}
    {"index":{"_index":"gen_rm_0"}}
    {"tag":-10.51,"category":"2025-08-02T18:54:38.328Z","result":{"gte":-2.63,"lte":42.07},"level":-68,"priority":[739.3,-162.39],"height":61.57,"address":737372,"value":{"gte":-922,"lte":8526},"price":-182,"size":810}
    {"index":{"_index":"gen_rm_0"}}
    {"tag":[-97.58,15.74],"category":"2024-09-20T10:23:20.268Z","result":{"gte":-0.36,"lte":7.529999999999999},"level":23,"status":"2021-04-03T23:50:25.583Z","priority":[986.35,-467.38],"height":-91.64,"value":{"gte":1435,"lte":8311}}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[82,38,24],"values":[-83.22,-78.72,-59.35]},"tag":-46.62,"category":"2023-01-31T00:11:04.280Z","result":{"gte":-46.69,"lte":9.160000000000004},"status":["2025-08-20T02:09:15.414Z","2021-12-19T17:14:16.731Z"],"priority":[-248.63,-290.66],"address":678767,"value":{"gte":-8151,"lte":-1215},"size":54}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[69,10,32],"values":[-24.55,-3.54,11.85]},"tag":-2.84,"category":"2022-09-06T14:24:44.282Z","level":-109,"status":"2022-04-07T06:15:18.820Z","height":79.84,"address":979309,"value":[{"gte":-9744,"lte":-8414},{"gte":-9645,"lte":-1530}],"price":-763,"size":408}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[35,55,35],"values":[-4.98,7.03,38.08]},"tag":25.66,"category":"2023-01-10T11:52:24.623Z","result":{"gte":-83.07,"lte":-5.099999999999994},"status":["2022-04-03T11:29:24.653Z","2025-03-20T05:57:54.971Z"],"priority":[278.92,-108.41],"price":214,"size":-227}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[63,21],"values":[-68.76,-28.9]},"tag":-52.79,"result":{"gte":17.32,"lte":45.89},"level":-107,"priority":[517.6,-50.76],"height":47.93,"address":616386,"value":{"gte":3380,"lte":7617},"price":-919,"size":-478}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[24,54,89,25],"values":[-93.41,-83.4,-66.99,-61.96]},"tag":86.34,"result":{"gte":28.35,"lte":89.45},"level":71,"status":"2024-12-23T05:42:35.580Z","priority":[[-805.14,-698.37],[260.15,-140.86]],"height":97.05,"address":538078,"price":841,"size":541}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[62,96,90,90,14,53],"values":[-8.24,38.94,41.06,59.94,78.45,112.38]},"tag":2.07,"result":{"gte":-93.84,"lte":-10.480000000000004},"level":[34,58],"status":"2024-05-05T06:44:47.162Z","height":68.18,"address":147189,"value":{"gte":-9935,"lte":-3176},"price":108,"size":[-225,331]}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[51,20,56,97,36,71],"values":[-20.54,-13.28,29.85,75.56,103.73,147.5]},"category":"2021-09-21T02:57:22.018Z","result":{"gte":-78.0,"lte":15.25},"level":-90,"status":"2023-06-14T19:36:30.999Z","priority":[873.58,61.54],"height":27.03,"address":690124,"value":[{"gte":-1329,"lte":912},{"gte":4927,"lte":7184}],"price":969,"size":-757}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[14,90],"values":[-67.34,-27.92]},"tag":-38.65,"category":["2024-12-28T21:17:35.076Z","2024-10-25T20:55:53.657Z"],"result":{"gte":-19.26,"lte":-12.470000000000002},"level":-69,"status":"2025-11-19T08:53:27.242Z","priority":[[461.18,-596.56],[474.42,419.44]],"height":44.96,"address":695359,"value":{"gte":2709,"lte":12022},"price":527,"size":-641}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[96,22,100,27,4],"values":[-11.19,26.64,32.17,35.91,74.54]},"category":"2023-11-25T09:22:15.665Z","result":{"gte":44.12,"lte":143.07999999999998},"level":-81,"status":"2022-03-01T14:01:20.586Z","priority":[[550.24,565.65],[635.29,-226.5]],"height":-4.09,"address":-63087,"value":[{"gte":-2874,"lte":-2853},{"gte":-8054,"lte":-3158}],"price":19}
    {"index":{"_index":"gen_rm_0"}}
    {"tag":-64.63,"category":"2021-02-25T18:07:17.432Z","level":23,"status":"2025-02-15T18:19:46.719Z","priority":[519.8,303.99],"height":-69.86,"address":-700112,"value":{"gte":-53,"lte":1917},"size":453}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[100,26,19],"values":[-60.23,-53.85,-50.85]},"tag":17.22,"category":"2021-04-21T05:06:06.228Z","result":{"gte":-76.96,"lte":-48.33999999999999},"level":[68,-27],"status":"2023-11-16T23:36:56.567Z","priority":[90.33,-381.6],"height":79.7,"address":-204658,"value":{"gte":-5251,"lte":-3501}}
    {"index":{"_index":"gen_rm_0"}}
    {"score":{"counts":[34,54,25,59],"values":[-40.85,-14.47,20.88,43.52]},"tag":22.53,"category":"2020-04-28T22:04:59.605Z","result":{"gte":-87.59,"lte":-43.92},"level":110,"status":"2023-01-14T22:42:48.013Z","priority":[-906.27,781.1],"height":32.65,"address":831860,"value":{"gte":-8219,"lte":-5264},"price":-150,"size":955}
    {"index":{"_index":"gen_rm_1"}}
    {"address":-227701,"count":800,"result":"134.129.1.131","height":-63.88,"active":-187978}
    {"index":{"_index":"gen_rm_1"}}
    {"address":-768134,"count":893,"result":"142.116.90.16","height":67.33,"color":687640,"weight":"delta fGBL","price":"bravo HVcCNkmqVJ","active":296901}
    {"index":{"_index":"gen_rm_1"}}
    {"address":107736,"count":221,"result":["6.55.200.222","94.179.125.252"],"height":-42.6,"color":836304,"weight":"delta Dnae","price":"delta aSkx","active":-330287}
    {"index":{"_index":"gen_rm_1"}}
    {"address":588359,"result":"6.194.115.191","height":0.02,"color":-771895,"weight":"delta EIZdfWAfSyX","price":"alpha mDPRDP","level":-86}
    {"index":{"_index":"gen_rm_1"}}
    {"address":-989237,"height":-8.28,"color":-795506,"weight":"charlie TVTd","active":113049}
    {"index":{"_index":"gen_rm_1"}}
    {"result":"153.34.193.255","height":[-82.49,-60.52],"color":-703304,"weight":["alpha AVmBOzNdYgoY","charlie mTSDQC"],"price":"alpha QBUUAl","level":49,"active":-898814}
    {"index":{"_index":"gen_rm_1"}}
    {"address":-937291,"count":-973,"result":"70.5.224.63","color":800721,"weight":"bravo uVIYelhrd","price":"delta rbwYjlfSL","level":19,"active":-489309}
    {"index":{"_index":"gen_rm_1"}}
    {"address":309122,"count":-517,"height":-35.15,"color":-409980,"price":"echo Eymq","level":-81,"active":-830417}
    {"index":{"_index":"gen_rm_1"}}
    {"address":370925,"result":"37.30.158.243","height":-73.61,"color":885920,"weight":"bravo rqOkBrcFp","price":"charlie KXfCJFQXeJ","active":172178}
    {"index":{"_index":"gen_rm_1"}}
    {"address":[259704,-329789],"count":681,"result":"174.9.221.200","color":[256178,-514928],"weight":"alpha CMLGygbVvlo","price":"bravo srwgwCt","level":-111,"active":-977424}
    {"index":{"_index":"gen_rm_1"}}
    {"address":327721,"result":["100.144.173.177","71.172.164.82"],"height":-14.58,"color":-50367,"price":"bravo Rekl","active":[22200,406386]}
    {"index":{"_index":"gen_rm_1"}}
    {"height":-7.58,"weight":"delta cNEXO","level":-3,"active":-746983}
    {"index":{"_index":"gen_rm_1"}}
    {"address":577222,"count":167,"result":"145.150.134.202","height":-47.77,"color":469732,"weight":"delta PfJ","price":"delta ybHYCQCsRpUp","active":34756}
    {"index":{"_index":"gen_rm_1"}}
    {"address":-437557,"count":911,"result":"234.210.15.94","height":17.83,"weight":["bravo tuUDoJco","charlie nKcAsmjw"],"level":-12,"active":-310322}
    {"index":{"_index":"gen_rm_1"}}
    {"address":[-120176,271108],"count":-532,"result":"127.184.161.88","height":-76.28,"color":-598704,"weight":["charlie gLhGasDPZGl","echo fWzpJ"],"price":"echo DFSQHTXGn","active":-128207}
    {"index":{"_index":"gen_rm_2"}}
    {"code":"2025-06-08T20:40:25.987358936Z","status":"2025-07-02T05:48:11.330Z","value":-841873,"host":[-33.95,59.37],"score":{"counts":[34,69,36,71],"values":[-28.62,-13.32,36.3,70.41]},"color":810104,"height":true,"tag":-6.36}
    {"index":{"_index":"gen_rm_2"}}
    {"price":["quick brown fox jumped","quick brown fox jumped"],"status":"2025-01-18T21:19:14.103Z","value":135786,"score":{"counts":[65,9,63,24,51,4],"values":[-66.34,-35.34,-20.28,-9.53,-3.12,29.41]},"color":-222536,"height":false,"tag":73.6}
    {"index":{"_index":"gen_rm_2"}}
    {"price":["single","quick brown fox jumped"],"code":"2025-08-01T01:02:55.135797190Z","status":["2020-06-19T17:03:21.433Z","2022-03-04T11:27:43.960Z"],"value":-842195,"score":{"counts":[81,15,56,85],"values":[-58.82,-31.11,-11.28,-2.19]},"color":-426527,"height":true}
    {"index":{"_index":"gen_rm_2"}}
    {"price":"one two three","code":"2022-05-29T07:32:17.589827687Z","status":"2021-07-08T08:05:16.890Z","value":575501,"host":-22.49,"score":{"counts":[58,69,9,72],"values":[-83.34,-54.08,-32.39,-19.6]},"weight":"delta ybwEUTxcYlu","color":[867159,-697390],"height":false}
    {"index":{"_index":"gen_rm_2"}}
    {"price":"hello","code":"2020-03-17T12:18:53.469308487Z","host":-90.97,"score":{"counts":[7,42,43,91],"values":[-62.77,-28.81,-12.59,2.17]},"weight":"alpha aaGodk","color":-959315,"height":true}
    {"index":{"_index":"gen_rm_2"}}
    {"code":"2024-04-08T06:22:24.319128797Z","value":-802236,"host":-20.88,"score":{"counts":[68,100,25,5],"values":[-51.99,-6.28,8.49,29.37]},"weight":"alpha pvoTG","color":490776,"height":false,"tag":1.54}
    {"index":{"_index":"gen_rm_3"}}
    {"weight":"charlie aVsqrbtVtqZA","category":"2025-12-24T05:43:50.237Z","ver":{"gte":"2020-06-14T14:24:45.119Z","lte":"2025-07-11T06:14:36.838Z"},"region":"alpha zCtDxhBdbF","count":656.5,"amount":498710,"path":-447,"name":[719.83,-956.95],"score":"random text here"}
    {"index":{"_index":"gen_rm_3"}}
    {"address":65.04,"weight":["alpha JJznLb","delta huDRds"],"category":["2022-01-29T20:43:32.805Z","2020-02-29T15:05:43.244Z"],"ver":{"gte":"2020-11-09T05:03:45.069Z","lte":"2021-05-30T03:56:13.251Z"},"region":"charlie rkOlc","rate":{"gte":"2024-11-09T02:02:06.352Z","lte":"2025-02-22T10:09:09.546Z"},"count":176.56,"code":"2020-11-28T20:50:13.518517460Z","amount":344788,"size":-817,"path":225,"name":[[762.96,454.78],[-741.79,-469.57]]}
    {"index":{"_index":"gen_rm_3"}}
    {"address":-31.44,"weight":"charlie MPlCfXV","category":"2021-02-07T21:20:48.829Z","ver":{"gte":"2020-12-09T20:16:42.200Z","lte":"2023-02-01T01:33:10.791Z"},"region":"delta BDLQJG","rate":{"gte":"2021-05-28T20:34:27.897Z","lte":"2023-09-07T00:15:43.782Z"},"count":929.09,"code":"2024-08-19T07:31:55.264721719Z","amount":165615,"distance":"2021-11-27T11:27:26.002Z","active":-813067,"size":-455,"path":436,"name":[-697.51,372.35],"score":"elastic search query"}
    {"index":{"_index":"gen_rm_3"}}
    {"ver":{"gte":"2021-05-22T05:39:32.694Z","lte":"2022-03-21T04:56:17.518Z"},"region":"delta SsrTecS","rate":{"gte":"2021-01-28T23:22:53.169Z","lte":"2023-06-24T23:54:10.571Z"},"count":148.58,"amount":[458126,731995],"distance":"2022-12-20T17:59:26.487Z","active":61765,"path":112,"name":[-643.83,-885.45],"score":"random text here"}
    {"index":{"_index":"gen_rm_3"}}
    {"address":49.24,"category":"2022-07-29T03:33:15.854Z","ver":{"gte":"2020-06-29T22:20:57.990Z","lte":"2020-09-14T04:25:55.075Z"},"region":"bravo JCwzXNBYYk","code":"2021-10-31T00:22:04.992324086Z","amount":467185,"distance":"2020-06-30T13:25:29.060Z","active":395909,"size":[-441,321],"path":-448,"name":[122.6,-804.79],"score":"random text here"}
    {"index":{"_index":"gen_rm_3"}}
    {"address":54.56,"ver":{"gte":"2023-04-07T21:43:54.415Z","lte":"2023-10-26T22:20:10.821Z"},"region":"bravo vaeMZzCXqoj","rate":{"gte":"2023-09-19T18:17:53.871Z","lte":"2025-12-10T17:20:00.809Z"},"count":464.65,"code":"2024-10-18T03:40:16.549697218Z","amount":261808,"distance":"2024-08-25T20:03:20.566Z","size":-737,"path":190,"name":[5.89,-311.08],"score":"hello world test"}
    {"index":{"_index":"gen_rm_3"}}
    {"address":-63.74,"category":"2025-12-17T12:07:07.474Z","ver":[{"gte":"2020-06-15T12:21:45.135Z","lte":"2024-06-29T07:45:49.795Z"},{"gte":"2022-08-15T16:31:59.860Z","lte":"2023-11-03T10:02:45.675Z"}],"rate":{"gte":"2021-10-23T06:25:43.363Z","lte":"2025-01-26T00:40:41.344Z"},"count":124.76,"code":"2024-09-25T01:48:49.883943058Z","amount":492563,"active":446037,"size":-608,"path":[-575,69],"score":"quick brown fox"}
    {"index":{"_index":"gen_rm_3"}}
    {"address":93.04,"weight":["alpha ZDWqFqwfrN","alpha EoWZslVLL"],"category":"2021-11-20T04:43:59.563Z","ver":{"gte":"2020-10-07T13:59:09.232Z","lte":"2023-08-17T15:27:57.014Z"},"region":"delta qlaxOT","code":"2023-04-15T19:56:04.116075806Z","amount":847292,"distance":"2024-02-15T18:12:35.596Z","active":338735,"size":209,"path":-222,"name":[963.77,-13.84]}
```

--- a/server/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java
+++ b/server/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java
@@ -29,6 +29,7 @@ import org.elasticsearch.index.fielddata.FieldDataContext;
 import org.elasticsearch.index.fielddata.IndexFieldData;
 import org.elasticsearch.index.fielddata.IndexFieldDataCache;
 import org.elasticsearch.index.fielddata.plain.BinaryIndexFieldData;
+import org.elasticsearch.index.mapper.blockloader.ConstantNull;
 import org.elasticsearch.index.mapper.blockloader.docvalues.DateRangeDocValuesLoader;
 import org.elasticsearch.index.query.SearchExecutionContext;
 import org.elasticsearch.indices.breaker.CircuitBreakerService;
@@ -354,13 +355,13 @@ public class RangeFieldMapper extends FieldMapper {
 
         @Override
         public BlockLoader blockLoader(BlockLoaderContext blContext) {
+            if (hasDocValues() == false) {
+                return ConstantNull.INSTANCE;
+            }
             if (rangeType != RangeType.DATE) {
                 throw new UnsupportedOperationException("loading blocks is only supported for date fields");
             }
-            if (hasDocValues()) {
-                return new DateRangeDocValuesLoader(name());
-            }
-            throw new IllegalStateException("Cannot load blocks without doc values");
+            return new DateRangeDocValuesLoader(name());
         }
     }
 

--- a/server/src/test/java/org/elasticsearch/index/mapper/DateRangeFieldMapperTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/DateRangeFieldMapperTests.java
@@ -16,6 +16,7 @@ import org.elasticsearch.common.breaker.CircuitBreaker;
 import org.elasticsearch.common.time.DateFormatter;
 import org.elasticsearch.common.time.DateUtils;
 import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.index.mapper.blockloader.ConstantNull;
 import org.elasticsearch.xcontent.ToXContent;
 import org.elasticsearch.xcontent.XContentBuilder;
 import org.junit.AssumptionViolatedException;
@@ -134,6 +135,22 @@ public class DateRangeFieldMapperTests extends RangeFieldMapperTests {
                 }
             }
         }
+    }
+
+    /**
+     * Test that block loader returns constant nulls when doc_values is disabled.
+     * Querying a date_range field with doc_values:false should produce null values
+     * rather than throwing an exception.
+     */
+    public void testNoDocValuesBlockLoader() throws IOException {
+        MapperService mapper = createSytheticSourceMapperService(mapping(b -> {
+            b.startObject("field");
+            b.field("type", "date_range");
+            b.field("doc_values", false);
+            b.endObject();
+        }));
+        BlockLoader loader = mapper.fieldType("field").blockLoader(new DummyBlockLoaderContext.MapperServiceBlockLoaderContext(mapper));
+        assertSame(ConstantNull.INSTANCE, loader);
     }
 
     @Override

--- a/server/src/test/java/org/elasticsearch/index/mapper/IntegerRangeFieldMapperTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/IntegerRangeFieldMapperTests.java
@@ -9,6 +9,7 @@
 
 package org.elasticsearch.index.mapper;
 
+import org.elasticsearch.index.mapper.blockloader.ConstantNull;
 import org.elasticsearch.xcontent.XContentBuilder;
 import org.junit.AssumptionViolatedException;
 
@@ -55,6 +56,38 @@ public class IntegerRangeFieldMapperTests extends RangeFieldMapperTests {
     @Override
     protected RangeType rangeType() {
         return RangeType.INTEGER;
+    }
+
+    /**
+     * Test that block loader returns constant nulls when doc_values is disabled for a non-DATE range type.
+     * This is the behavior after reordering the checks in {@code blockLoader}: doc_values is checked before
+     * range type, so a non-DATE range with doc_values disabled returns null rather than throwing.
+     */
+    public void testNoDocValuesBlockLoader() throws IOException {
+        MapperService mapper = createSytheticSourceMapperService(mapping(b -> {
+            b.startObject("field");
+            b.field("type", "integer_range");
+            b.field("doc_values", false);
+            b.endObject();
+        }));
+        BlockLoader loader = mapper.fieldType("field").blockLoader(new DummyBlockLoaderContext.MapperServiceBlockLoaderContext(mapper));
+        assertSame(ConstantNull.INSTANCE, loader);
+    }
+
+    /**
+     * Test that a non-DATE range type with doc_values enabled still throws UnsupportedOperationException,
+     * since only date_range supports block loading via doc values.
+     */
+    public void testBlockLoaderWithDocValues() throws IOException {
+        MapperService mapper = createSytheticSourceMapperService(mapping(b -> {
+            b.startObject("field");
+            b.field("type", "integer_range");
+            b.endObject();
+        }));
+        expectThrows(
+            UnsupportedOperationException.class,
+            () -> mapper.fieldType("field").blockLoader(new DummyBlockLoaderContext.MapperServiceBlockLoaderContext(mapper))
+        );
     }
 
     @Override

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
@@ -292,6 +292,28 @@ public abstract class FieldExtractorTestCase extends ESRestTestCase {
         new Test("version").test(randomVersionString());
     }
 
+    /**
+     * Querying a date_range field with doc_values disabled should return null for every document
+     * rather than throwing an exception. Before the fix, this caused a full query failure because
+     * the block loader tried to access doc values that were not present.
+     */
+    public void testDateRangeNoDocValues() throws IOException {
+        assumeDateRangeFieldTypeSupported();
+        createIndex("test", index -> {
+            index.startObject("properties");
+            index.startObject("dates");
+            index.field("type", "date_range");
+            index.field("doc_values", false);
+            index.endObject();
+            index.endObject();
+        });
+        index("test", """
+            {"dates": {"gte": "2024-01-01", "lte": "2024-12-31"}}""");
+
+        Map<String, Object> result = runEsql("FROM test* | LIMIT 10");
+        assertResultMap(result, List.of(columnInfo("dates", "date_range")), List.of(matchesList().item(null)));
+    }
+
     public void testGeoPoint() throws IOException {
         new Test("geo_point")
             // TODO we should support loading geo_point from doc values if source isn't enabled
@@ -1520,6 +1542,12 @@ public abstract class FieldExtractorTestCase extends ESRestTestCase {
         var capsName = EsqlCapabilities.Cap.REPORT_ORIGINAL_TYPES.name().toLowerCase(Locale.ROOT);
         boolean requiredClusterCapability = clusterHasCapability("POST", "/_query", List.of(), List.of(capsName)).orElse(false);
         assumeTrue("This test makes sense for versions that report original types", requiredClusterCapability);
+    }
+
+    private void assumeDateRangeFieldTypeSupported() throws IOException {
+        var capsName = EsqlCapabilities.Cap.DATE_RANGE_FIELD_TYPE_V2.name().toLowerCase(Locale.ROOT);
+        boolean requiredClusterCapability = clusterHasCapability("POST", "/_query", List.of(), List.of(capsName)).orElse(false);
+        assumeTrue("date_range field type not supported in this version", requiredClusterCapability);
     }
 
     private void assumeSuggestedCastReported() throws IOException {

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
@@ -1545,7 +1545,7 @@ public abstract class FieldExtractorTestCase extends ESRestTestCase {
     }
 
     private void assumeDateRangeFieldTypeSupported() throws IOException {
-        var capsName = EsqlCapabilities.Cap.DATE_RANGE_FIELD_TYPE_V2.name().toLowerCase(Locale.ROOT);
+        var capsName = EsqlCapabilities.Cap.DATE_RANGE_FIELD_TYPE_V3.name().toLowerCase(Locale.ROOT);
         boolean requiredClusterCapability = clusterHasCapability("POST", "/_query", List.of(), List.of(capsName)).orElse(false);
         assumeTrue("date_range field type not supported in this version", requiredClusterCapability);
     }

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRandomMappingRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRandomMappingRestTest.java
@@ -58,11 +58,6 @@ public abstract class GenerativeRandomMappingRestTest extends GenerativeRestTest
 
     private static volatile List<GeneratedIndex> generatedIndices;
 
-    private static final Pattern CANNOT_LOAD_BLOCKS_WITHOUT_DOC_VALUES = Pattern.compile(
-        ".*Cannot load blocks without doc values.*",
-        Pattern.DOTALL
-    );
-
     // FORK converts UnsupportedAttribute to a plain ReferenceAttribute(UNSUPPORTED), stripping the
     // Unresolvable marker. Functions then reject the argument with a generic "type [unsupported]" error
     // instead of the proper "Cannot use field" message. https://github.com/elastic/elasticsearch/issues/147094
@@ -226,11 +221,6 @@ public abstract class GenerativeRandomMappingRestTest extends GenerativeRestTest
 
     private static boolean isAdditionalAllowedError(String errorMessage, String query) {
         if (errorMessage == null) return false;
-        // RangeFieldMapper throws "Cannot load blocks without doc values" for *_range fields
-        // with doc_values:false. https://github.com/elastic/elasticsearch/issues/146527
-        if (isAllowedError(errorMessage, CANNOT_LOAD_BLOCKS_WITHOUT_DOC_VALUES) && hasRangeFieldType()) {
-            return true;
-        }
         // FORK converts UnsupportedAttribute to ReferenceAttribute(UNSUPPORTED), causing functions
         // to reject the field with a generic "type [unsupported]" error.
         // https://github.com/elastic/elasticsearch/issues/147094
@@ -242,21 +232,6 @@ public abstract class GenerativeRandomMappingRestTest extends GenerativeRestTest
         // https://github.com/elastic/elasticsearch/issues/147610
         if (query != null && isAllowedError(errorMessage, FAILED_TO_CREATE_QUERY) && queryContainsFullTextFunction(query)) {
             return true;
-        }
-        return false;
-    }
-
-    private static boolean hasRangeFieldType() {
-        List<GeneratedIndex> indices = generatedIndices;
-        if (indices == null) {
-            return false;
-        }
-        for (GeneratedIndex idx : indices) {
-            for (RandomMappingGenerator.FieldDef field : idx.fields()) {
-                if (field.esType().endsWith("_range")) {
-                    return true;
-                }
-            }
         }
         return false;
     }

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java
@@ -1797,8 +1797,9 @@ public class EsqlCapabilities {
 
         /**
          * Support for the DATE_RANGE field type, RANGE_WITHIN, TO_DATE_RANGE(string), RANGE_MIN, RANGE_MAX.
+         * V3: DATE_RANGE fields with {@code doc_values: false} now return null instead of throwing an exception.
          */
-        DATE_RANGE_FIELD_TYPE_V2(Build.current().isSnapshot()),
+        DATE_RANGE_FIELD_TYPE_V3(Build.current().isSnapshot()),
 
         /**
          * Network direction function.

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java
@@ -4406,7 +4406,7 @@ public class StatementParserTests extends AbstractStatementParserTests {
         try (XContentBuilder report = JsonXContent.contentBuilder().humanReadable(true).prettyPrint().lfAtEnd()) {
             report.startObject();
             List<String> namesAndAliases = new ArrayList<>(DataType.namesAndAliases());
-            if (EsqlCapabilities.Cap.DATE_RANGE_FIELD_TYPE_V2.isEnabled() == false) {
+            if (EsqlCapabilities.Cap.DATE_RANGE_FIELD_TYPE_V3.isEnabled() == false) {
                 // Some types do not have a converter function if the capability is disabled
                 namesAndAliases.removeAll(List.of("date_range"));
             }

